rllib/env/single_agent_env_runner.py

```diff
@@ -229,6 +230,7 @@ def _sample_timesteps(
             # Reset the environment.
             # TODO (simon): Check, if we need here the seed from the config.
             obs, infos = self.env.reset()
+            obs = unbatch(obs)
```

Comment:
Needed for nested obs spaces.

Reply:
Sweet. This makes `Dict` and `Tuple` spaces now possible.

Reply:
Yes! :)

Reply:
I am thinking in this regard about using in all algorithms mainly the base encoder b/c this enables different  input formats automatically. Using then e.g. in `SAC` for `SampleBatch.ACTIONS` an extra encoder and aggregate the two streams together. Like this all kind of observations are handled by the base catalog.